### PR TITLE
Add join result fields we likely require in our analysis

### PR DESCRIPTION
**Pull request by amirfarhat**
Per join result:
* Add table-specific fields (records, columns) so data processor can plot ratio of table size
* Add size of emitted records to measure so data processor can plot join specificity

@parama you might find these more helpful for data plotting 😃

### CODE REVIEW

File: benchmark_joins/src/runner.rs [hunk 1]
```diff
@@ -9,12 +9,20 @@ use crate::NestedLoopsJoin;
 use crate::SimpleTable;
 use crate::JoinAlgos;
 
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Table {
+	table_name: String,
+	num_records: usize,
+  columns_per_record: usize,
```
Comment (dmorejon) [inline benchmark_joins/src/runner.rs:16]: nit spacing

File: benchmark_joins/src/runner.rs [hunk 2]
```diff
@@ -9,12 +9,20 @@ use crate::NestedLoopsJoin;
...
+
 #[derive(Serialize, Deserialize, Debug)]
 pub struct JoinRunResult {
 	join_type: JoinAlgos,
 	execution_time_nanos: u128,
-	outer_table: String,
-	inner_table: String,
+	outer_table: Table,
+	inner_table: Table,
+  num_emitted_records: usize,
```
Comment (dmorejon) [inline benchmark_joins/src/runner.rs:25]: nit spacing

File: benchmark_joins/src/runner.rs [hunk 3]
```diff
@@ -30,20 +38,38 @@ fn flush_caches() -> () {
...
+    inner_table: t2,
+    num_emitted_records: results.len(),
+  }
 }
 
-fn run_nl_join(nlj: &mut NestedLoopsJoin, left_col: usize, right_col: usize) -> Duration {
-	flush_caches();
+fn run_nl_join(nlj: &mut NestedLoopsJoin, left_col: usize, right_col: usize, t1: Table, t2: Table) -> JoinRunResult {
+	// Run the join
+  flush_caches();
```
Comment (dmorejon) [inline benchmark_joins/src/runner.rs:60]: nit spacing

File: benchmark_joins/src/runner.rs [hunk 4]
```diff
@@ -30,20 +38,38 @@ fn flush_caches() -> () {
 	}
 }
 
-fn run_bnl_join(bnlj: &mut BlockNL, left_col: usize, right_col: usize) -> Duration {
-	flush_caches();
+fn run_bnl_join(bnlj: &mut BlockNL, left_col: usize, right_col: usize, t1: Table, t2: Table) -> JoinRunResult {
+	// Run the join
+  flush_caches();
```
Comment (dmorejon) [inline benchmark_joins/src/runner.rs:43]: nit spacing

**Review by dmorejon — APPROVED** [review]

File: benchmark_joins/src/runner.rs (hunk 4 above)
Comment (amirfarhat) [reply benchmark_joins/src/runner.rs]: Resolved

File: benchmark_joins/src/runner.rs (hunk 3 above)
Comment (amirfarhat) [reply benchmark_joins/src/runner.rs]: Resolved

File: benchmark_joins/src/runner.rs (hunk 2 above)
Comment (amirfarhat) [reply benchmark_joins/src/runner.rs]: Resolved

File: benchmark_joins/src/runner.rs (hunk 1 above)
Comment (amirfarhat) [reply benchmark_joins/src/runner.rs]: Resolved

**Pull request merged**